engine/docker/docker.go

```diff
@@ -525,3 +537,33 @@ func (e *Docker) setProxyInDockerDaemon(ctx context.Context, pipelineConfig *spe
 		}
 	}
 }
+
+// SoftStop stops the container giving them a 30 seconds grace period. The signal sent by ContainerStop is SIGTERM.
+// After the grace period, the container is killed with SIGKILL.
+// After all the containers are stopped, they are removed only when the status is not "running" or "removing".
+func (e *Docker) SoftStop(ctx context.Context, name string) {
+	timeout := 30 * time.Second
+	if err := e.client.ContainerStop(ctx, name, &timeout); err != nil {
+		logrus.WithField("container", name).WithField("error", err).Warnln("failed to stop the container")
```

Comment:
if we haven't been able to stop the contain we should probably return as there's no point in continuing with the rest of the procedure.

Reply:
This will run for all containers, the ones that are already stopped because they were finalized too. Which means that they will return with err, which is why we don't return. We want to delete those containers from the docker daemon. 
Also for other containers that we can't stop, removing them will send a kill -9 ensuring that they are cleaned